hooks/useShowByScroll.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react';
+
+import useScrollPosition from './useScrollController';
+import { useScrollDirection } from './useScrollDirection';
+
+export default function useShowByScroll() {
+  const [showBottom, setShowBottom] = useState(true);
+  const scrollDirection = useScrollDirection();
+  const { position } = useScrollPosition();
+
+  let timerId: ReturnType<typeof setTimeout>;
```

Comment:
```suggestion
    const timerId = useRef<ReturnType<typeof setTimeout> | null>(null);
  const debounce = (func: Function, sec: number) => {
    if (timerId.current) clearTimeout(timerId.current);
    timerId.current = setTimeout(() => func(), sec);
  };
```
최근에 알게되었는데 요렇게 타이머를 useRef로 선언해야 clear가 잘 되더라구용